torch/jit/__init__.py

```diff
@@ -581,9 +582,16 @@ def __getattr__(self, attr):
         return self.module._get_method(attr)
 
 
-def script(fn, optimize=True, _frames_up=0):
+def weak_script(fn):
+    fn._jit_is_weak_script = True
```

Comment:
We should probably not modify functions directly. This can do:

```
compiled_weak_fns[fn] = "pending"
```

And then we can have a function:

```
def _try_compile_weak_script(fn):
    v =  compiled_weak_fns.get(fn)
   if v is None: 
      return None
   if v == "pending"
      v = torch.jit.script(fn)
      compiled_weak_fns[fn] = v
   return v
```